app/services/workflow_executions/completion_service.rb

```diff
@@ -112,9 +117,32 @@ def process_samples_metadata(run_output_data:)
       run_output_data['metadata']['samples']&.each do |sample_puid, sample_metadata|
         # This assumes the sample puid matches, i.e. happy path
         samples_workflow_execution = get_samples_workflow_executions_by_sample_puid(puid: sample_puid)
-        samples_workflow_execution.metadata = sample_metadata
+        samples_workflow_execution.metadata = flatten(sample_metadata)
         samples_workflow_execution.save!
       end
     end
+
+    def merge_metadata_onto_samples
+      @workflow_execution.samples_workflow_executions&.each do |swe|
+        next if swe.metadata.nil?
+
+        params = { 'metadata' => swe.metadata }
```

Comment:
**nitpick (blocking):** Can you add `analysis_id` to the params hash and set it to `@workflow_execution.id`. This way we can track that it was the `workflow_execution` that wrote this metadata.

Reply:
added in https://github.com/phac-nml/irida-next/pull/504/commits/fecd6a50f494346e944bd756a5f2bfbd04c9579e